### PR TITLE
drivers: serial: Update cc32xx uart interrupt handling

### LINKED ISSUE #11202: recent "shell: shell_uart" commit breaks shell samples on cc3220sf
The commit: 4962618e "shell: shell_uart: add ring_buffers and interrupt support" breaks the samples/subsys/shell/shell_module and samples/net/wifi shell samples on the cc3220sf_launchxl, resulting in no character input.

Reverting that commit makes the samples work again.

The interrupt driven uart tests had previously been working, so there is some assumption being made with this new shell client that isn't being met by the cc32xx_uart.

### PR DESCRIPTION
**Pull request by GAnthony**
The Zephyr console and shell interrupt processing assumes
a TX interrupt is evoked upon first enabling the TX
interrupt via uart_irq_tx_enable.

This was not the case with the cc32xx uart, coming out of
reset, with FIFO's disabled.

The only way found to achieve this behavior is to fill
the fifo with a non-printable character on initialization.

Also, the uart driver was explicitly clearing TX/RX interrupts in
its isr, which was unnecessary, as the act of reading/writing
did that implicitly.

These fixes allow the cc32xx uart to work with the
current Zephyr console/shell design.

Fixes: #11202

### CODE REVIEW

File: drivers/serial/uart_cc32xx.c [hunk 1]
```diff
@@ -70,15 +72,21 @@ static int uart_cc32xx_init(struct device *dev)
...
 	IRQ_CONNECT(DT_TI_CC32XX_UART_4000C000_IRQ_0,
 		    DT_TI_CC32XX_UART_4000C000_IRQ_0_PRIORITY,
 		    uart_cc32xx_isr, DEVICE_GET(uart_cc32xx_0),
 		    0);
 	irq_enable(DT_TI_CC32XX_UART_4000C000_IRQ_0);
+
+	/* Fill the tx fifo, so Zephyr console & shell subsystems get "primed"
+	 * with first tx fifo empty interrupt when they first call
+	 * uart_irq_tx_enable().
+	 * Use the BEL character, which typically prints nothing to the console.
```
Comment (pfalcon) [inline drivers/serial/uart_cc32xx.c]: Sorry, but BEL "typically" produces a sound. If someone really has such classical setup, and Zephyr rings on each boot to them, they will soon hate Zephyr ;-).

**Review by pfalcon — COMMENTED** [review]
Suggested to use safer "priming" char. If no other ideas, '\r' would do.

File: drivers/serial/uart_cc32xx.c (hunk 1 above)
Comment (GAnthony) [reply drivers/serial/uart_cc32xx.c]: It all depends on the terminal emulator - most don't ring a bell anymore.  I haven't found one that does. 
But I can change it to '\r'.

**Comment by GAnthony** [comment]
@pfalcon, updated per request.
Should be ready to merge.

**Review by pfalcon — APPROVED** [review]
Thanks for updates!

**Comment by pfalcon** [comment]
@GAnthony, By a chance, do you follow discussion with @fvincenzo in https://github.com/zephyrproject-rtos/zephyr/pull/11314 ? You likely would need to make a similar change (once it's decided what change exacty needs to be done).

**Comment by pfalcon** [comment]
> You likely would need to make a similar change (once it's decided what change exacty needs to be done).

Actually no, as we settled (so far) on implicit acknowledge by FIFO ops. Sorry, have my head dizzy following too much stuff at once. Still, following that discussion makes sense, as we should build up common knowledge of how to write UART drivers correctly for various hw cases.

**Comment by pfalcon** [comment]
@jarz-nordic: This fixes shell handling for cc32xx, so please approve.

**Comment by GAnthony** [comment]
> 
> 
> @GAnthony, By a chance, do you follow discussion with @fvincenzo in #11314 ? ...

Yes, it was a similar confusion I had (before benefiting from your explanation of the uart API contract) since I would say most driver writers are taught to acknowledge interrupts immediately in the ISR.

**Comment by jarz-nordic** [comment]
Its fine :)

**Pull request merged**